router/core/src/main/scala/io/buoyant/router/Router.scala

```diff
@@ -286,7 +291,7 @@ object StackRouter {
 
     case class PathParams(prefix: PathMatcher, mk: Map[String, String] => Stack.Params)
 
-    case class PerPathParams(params: Seq[PathParams]) {
+    case class PerPathParams(params: Seq[PathParams], dynamicParams: Option[Activity[Seq[PathParams]]]) {
```

Comment:
It's worth considering if we can just have a `Activity[Seq[PathParams]]`.  The static and global configs could simply return an `Activity.value(...)` that never changes.